--- a/src/cli/libnvc.c
+++ b/src/cli/libnvc.c
@@ -100,9 +100,9 @@
                 funcs_entry(version),
         };
 
-        const size_t len_libnvc = offsetof(struct libnvc, nvcaps_style)/sizeof(void*);
-        const size_t len_funcs = sizeof(funcs)/sizeof(*funcs);
-        static_assert(len_funcs == len_libnvc, "len(libnvc) != len(funcs)");
+#define len_libnvc offsetof(struct libnvc, nvcaps_style)/sizeof(void*)
+#define len_funcs sizeof(funcs)/sizeof(*funcs)
+        static_assert((len_funcs) == (len_libnvc), "len(libnvc) != len(funcs)");
 
         void *handle = dlopen(libnvc_v0_soname, RTLD_NOW);
         if (!handle) {
